api.proto

```diff
@@ -3,13 +3,28 @@ package proto;
 
 option java_package = "io.liftbridge.proto";
 
+// Custom Stream Config
+enum   CompactEnabled {     
+    UNKNOWN  = 0;
+    ENABLED  = 1;
+    DISABLED = 2;
+}
+
 // CreateStreamRequest is sent to create a new stream.
 message CreateStreamRequest {
-    string subject           = 1; // Stream NATS subject
-    string name              = 2; // Stream name (unique per subject)
-    string group             = 3; // Partitions NATS subject amongst group members
-    int32  replicationFactor = 4; // Number of stream replicas
-    int32  partitions        = 5; // Number of stream partitions
+    string         subject              = 1; // Stream NATS subject
+    string         name                 = 2; // Stream name (unique per subject)
+    string         group                = 3; // Partitions NATS subject amongst group members
+    int32          replicationFactor    = 4; // Number of stream replicas
+    int32          partitions           = 5; // Number of stream partitions
+	int64          RetentionMaxBytes    = 6; // Custom StreamConfig
```

Comment:
Thinking about this, don't we have a similar problem as we had with knowing whether stream compaction was set? I.e., if we assume a value of 0 indicates no limit, what about the scenario where the server has a limit of 1024 and the client doesn't explicitly set this value. The value on the request will be 0, but the client intended to use the server default.

Reply:
Indeed. Now I am a bit nervous. That's a fair point, since most default values are 0,  in this PR
on server side, it effectively discart `0` [value](https://github.com/liftbridge-io/liftbridge/blob/6b3f207c3bbfef948f7124fbff2cb128c53db724/server/config.go#L185)

That means for `RetentionMaxBytes`, there is no way to customly set `0` to indicate no limits and client has to rely on serveri's default config.

Should we use `enum` for all of these configuration ? It means more refactoring, but to be on safe side.

Reply:
Do you think we need to use the same pattern as `CompactEnabled`? I hate the idea of having to have an enum _and_ value field for each config, but I'm struggling to think of a better way to distinguish between set/not set/server default.

Reply:
I am still thinking but it seems we would need something like

```
enum RetentionMaxBytesOverwrite {
    DEFAULT = 0;
    ENABLED  = 1;
    DISABLED = 2;
}
``` 

And the server side will check this enum type before overwrite the `RetentionMaxBytes` configuration. It looks heavy to be honest.


Or we can change the server configuration to consider `-1` as unlimited instead of 0. It will be change that may affects older version, but it seems logically to say `-1` as unlimited.

What do you think ?

Reply:
Unfortunately, I don't think we can change to -1 since that would break existing behavior in a bad way. I'm going to think on it a bit more...

Reply:
Another possible solution is wrapping the field in a `message`, which allows it to be nullable. So not setting it would leave it `null`, which would indicate using the server default. Something like:

```protobuf
message NullableInt32 {
    int32 value = 1;
}
```

Not pretty, and it would require a separate wrapper for each type.

Could also potentially hack something with [`map`](https://developers.google.com/protocol-buffers/docs/proto3#maps) which would allow for generics or possibly [`oneof`](https://developers.google.com/protocol-buffers/docs/proto3#oneof). None seem like an ideal solution.

Reply:
In fact, the only safe and less ambiguous  way is indeed using a wrapper, each setting will be wrapped in a `message`, which clearly indicates a value, else the whole setting message will be discarded.

I would say it is rather heavy but I would not say that it is not clean. In fact I consider it to be a clean solution, since it guarantees explicit configuration setting, no ambiguous value using out of the box basic type. Yet a bit extra heavy to cannot say it's bad.

Indeed, speak it out loud, changing configuration back to `-1` is a dirty hack..

Not ideal, but as long as it is the next-best-thing that can be done without introducing ambiguity  or bad pattern in the code base, should we just go ahead for it ?

Reply:
I agree, I think it's our best option.

Reply:
OK. Then let me spend time to re-write the code to use this for 

```
retention.max.bytes
retention.max.messages
```

How's about `compact.enabled`, it is already an `enum`, but just for keeping everything homogeneous, should we use the above approach for `compact.enabled` as well ?

Reply:
Yeah, I think we should keep everything consistent.

Reply:
The changes have been implemented. 
`RetentionMaxBytes`, `RetentionMaxMessages`  and `CompactEnabled` now use embedded message .